ledger/src/shred/merkle.rs

```diff
@@ -780,41 +688,42 @@ fn make_merkle_proof(
 }
 
 pub(super) fn recover(
-    shreds: Vec<Shred>,
+    mut shreds: Vec<Shred>,
     reed_solomon_cache: &ReedSolomonCache,
 ) -> Result<impl Iterator<Item = Shred>, Error> {
-    // Grab {common, coding} headers from first coding shred.
+    // Sort shreds by their erasure shard index.
+    // In particular this places all data shreds before coding shreds.
```

Comment:
Could we get away without the sort here? If I understand correctly the blockstore code already ensures that the shreds are sorted by index / type here https://github.com/anza-xyz/agave/blob/e8463351d94a49543e00ba8eee559f29b93bd0ed/ledger/src/blockstore.rs#L833